django/contrib/postgres/indexes.py

```diff
@@ -36,6 +36,44 @@ def get_with_params(self):
         return []
 
 
+class BloomIndex(PostgresIndex):
+    suffix = 'bloom'
+
+    def __init__(self, *, length=None, columns=None, **kwargs):
+        if length is not None and not 0 < length <= 4096:
+            raise ValueError(
+                'length must be None or a positive integer up to a maximum of 4096.'
+            )
+        if columns and not (len(columns) <= 32 and all(0 < v <= 4095 for v in columns)):
+            raise ValueError(
+                'columns must be None or an iterable of up to 32 positive '
+                'integers each up to a maximum of 4095.'
+            )
```

Comment:
All fields have default entries (2 bits) in columns, that creates a similar limitation for the list of fields, e.g.
```python
django.contrib.postgres.indexes.BloomIndex(fields=[
    'field_int_1', 'field_int_2', 'field_int_3', 'field_int_4', 'field_int_5', 'field_int_6', 'field_int_7',
    'field_int_8', 'field_int_9', 'field_int_10', 'field_int_11', 'field_int_12', 'field_int_13',
    'field_int_14', 'field_int_15', 'field_int_16', 'field_int_17', 'field_int_18', 'field_int_19',
    'field_int_20', 'field_int_21', 'field_int_22', 'field_int_23', 'field_int_24', 'field_int_25',
    'field_int_26', 'field_int_27', 'field_int_28', 'field_int_29', 'field_int_30', 'field_int_31',
    'field_int_32', 'field_int_33',
], name='test_bloom_index'),
```
will fail with 
```
django.db.utils.OperationalError: cannot use more than 32 columns in an index
```

Reply:
Ah, yes!